executor/builder.go

```diff
@@ -650,14 +650,38 @@ func (b *executorBuilder) buildSelectLock(v *plannercore.PhysicalLock) Executor
 		}
 	}
 
-	if len(e.partitionedTable) > 0 {
-		schema := v.Schema()
-		e.tblID2PIDColumnIndex = make(map[int64]int)
-		for i := 0; i < len(v.ExtraPIDInfo.Columns); i++ {
-			col := v.ExtraPIDInfo.Columns[i]
-			tblID := v.ExtraPIDInfo.TblIDs[i]
-			offset := schema.ColumnIndex(col)
-			e.tblID2PIDColumnIndex[tblID] = offset
+	if len(v.PartitionedTable) > 0 {
+		e.partitionedTable = make(map[int64]table.PartitionedTable)
+		e.tblID2PtColsOffsets = make(map[int64][]int)
+
+		for _, pt := range v.PartitionedTable {
+			tblInfo := pt.Meta()
+			e.partitionedTable[tblInfo.ID] = pt
+			dbInfo, ok := is.SchemaByTable(tblInfo)
+			if !ok {
+				b.err = errors.Trace(errors.Errorf("Cannot get schema info for table %s", tblInfo.Name.O))
+				return nil
+			}
+
+			cols := pt.VisibleCols()
+			offsets := make([]int, 0, len(cols))
+			for _, colInfo := range cols {
+				colName := fmt.Sprintf("%s.%s.%s", dbInfo.Name.L, tblInfo.Name.L, colInfo.Name.L)
```

Comment:
Will there be some colume names not in this format like `_tidb_rowid` that it could not be found in the schema？